### PR TITLE
feat: add debug builds for CI

### PR DESCRIPTION
**Pull request by EmosewaMC**
alongside that, fixes the issues caused by adding said pipelines
(copying over the shared libraries to directories where they are expected to exist on windows and mac)

### CODE REVIEW

File: CMakePresets.json [hunk 1]
```diff
@@ -5,124 +5,163 @@
...
-			"jobs": 2
-		},
-        "output": {
-          "outputOnFailure": true
-        },
-		"filter": {
-			"exclude": {
-				"name": "((example)|(minigzip))+"
-			}
+  "configurePresets": [
```
Comment (EmosewaMC) [inline CMakePresets.json:8]: We had to do the whitespace changes at somepoint sorry 🙈

**Review by aronwk-aaron — APPROVED** [review]

**Comment by jadebenn** [comment]
This isn't quite configured correctly, I don't think. MSVC builds are handled a bit differently and (generally) shouldn't require separate configurations for debug and release because the MSVC build system is smart enough to be able to read the build preset instead.

**Comment by jadebenn** [comment]
Screenshot 2024-11-16 18 44 46

**Comment by jadebenn** [comment]
Since we merged #1439, one thing I think that might be worth further research on your end is whether we need to keep some of these MSVC presets. On my end, both 'Windows Only Configure Settings' and 'Default' don't work properly.
image

**Pull request closed without merging**